salt/utils/dns.py

```diff
@@ -1216,5 +1204,67 @@ def parse_resolv(src="/etc/resolv.conf"):
             "search": search,
             "options": options,
         }
-    except IOError:
+    except OSError:
+        return {}
+
+
+def parse_systemd_resolve():
+    """
+    Parse the information from systemd-resolve
+    """
+
+    nameservers = []
+    ip4_nameservers = []
+    ip6_nameservers = []
+    domain = ""
+
+    cmd = "systemd-resolve --status"
+    cmd = __salt__["cmd.run_all"](
+        "{}".format(cmd), python_shell=False, output_loglevel="quiet"
+    )
+
+    if cmd["retcode"] == 0:
+        prev_section = None
+        lines = [re.sub("[\t ]", "", line) for line in cmd["stdout"].splitlines()]
+
+        contents = {}
+        prev_section = None
+        for line in lines:
+            section = None
+            if ":" in line and line.count(":") <= 1:
+                section, value = line.split(":")
+            else:
+                value = line
+            if not section:
+                section = prev_section
+
+            if section:
+                if section in contents:
+                    contents[section].append(value)
+                else:
+                    contents[section] = [value]
+                prev_section = section
+
+        for item in contents:
+            if item == "DNSServers":
+                for addr in contents[item]:
+                    try:
+                        ip_addr = ipaddress.ip_address(addr)
+                        version = ip_addr.version
+                        ip_addr = str(ip_addr)
+                        if ip_addr not in nameservers:
+                            nameservers.append(ip_addr)
+                        if version == 4 and ip_addr not in ip4_nameservers:
+                            ip4_nameservers.append(ip_addr)
+                        elif version == 6 and ip_addr not in ip6_nameservers:
+                            ip6_nameservers.append(ip_addr)
+                    except ValueError as exc:
+                        log.error("%s: %s", addr, exc)
```

Comment:
This usually hides valuable information on the traceback. Let's at least add `exc_info_on_loglevel=logging.DEBUG`

Reply:
👍